core/src/test/java/org/apache/iceberg/TestMetadataTableScans.java

```diff
@@ -63,6 +67,28 @@ private void preparePartitionedTable() {
         .commit();
   }
 
+  @Test
+  public void testManifestsTableWithDroppedPartition() throws IOException {
+    table.newFastAppend()
+        .appendFile(FILE_A)
+        .appendFile(FILE_B)
+        .commit();
+
+    table.updateSpec().removeField(Expressions.bucket("data", 16)).commit();
+    table.refresh();
+
+    Table manifestsTable = new ManifestsTable(table.ops(), table);
+    TableScan scan = manifestsTable.newScan()
+            .filter(Expressions.lessThan("length", 10000L));
```

Comment:
Why are we filtering?

Reply:
It could be removed. The test is adjusted from `testManifestsTableAlwaysIgnoresResiduals`.